tests/test_multi_mesh.cpp

```diff
@@ -16,85 +18,263 @@ constexpr PrimitiveType PV = PrimitiveType::Vertex;
+        auto tups = parent.map_to_child_tuples(mesh, Simplex(PF, tuple));
+        REQUIRE(tups.size() == 1);
+        return tups[0];
+    };
 
     Tuple edge = parent.edge_tuple_between_v1_v2(1, 0, 0);
-    Tuple edge_child0 = MultiMeshManager::map_tuple_between_meshes(parent, *child0_ptr, parent.multi_mesh_manager.map_to_child_handles[0], edge);
-    Tuple edge_child1 = MultiMeshManager::map_tuple_between_meshes(parent, *child1_ptr, parent.multi_mesh_manager.map_to_child_handles[1], edge);
-    Tuple edge_child2 = MultiMeshManager::map_tuple_between_meshes(parent, *child2_ptr, parent.multi_mesh_manager.map_to_child_handles[2], edge);
+    Tuple edge_child0 = get_single_child_tuple(child0, edge);
+    Tuple edge_child1 = get_single_child_tuple(child1, edge);
+    Tuple edge_child2 = get_single_child_tuple(child2, edge);
 
-    CHECK(edge_child0 == child0_ptr->edge_tuple_between_v1_v2(1, 0, 0));
-    CHECK(edge_child1 == child1_ptr->edge_tuple_between_v1_v2(1, 0, 0));
+    CHECK(edge_child0 == child0.edge_tuple_between_v1_v2(1, 0, 0));
+    CHECK(edge_child1 == child1.edge_tuple_between_v1_v2(1, 0, 0));
     CHECK(edge_child2 == child2_ptr->edge_tuple_between_v1_v2(1, 0, 0));
 
-    CHECK(child0_ptr->switch_vertex(edge_child0) == MultiMeshManager::map_tuple_between_meshes(parent, *child0_ptr, parent.multi_mesh_manager.map_to_child_handles[0], parent.switch_vertex(edge)));
-    CHECK(child1_ptr->switch_vertex(edge_child1) == MultiMeshManager::map_tuple_between_meshes(parent, *child1_ptr, parent.multi_mesh_manager.map_to_child_handles[1], parent.switch_vertex(edge)));
-    CHECK(child2_ptr->switch_vertex(edge_child2) == MultiMeshManager::map_tuple_between_meshes(parent, *child2_ptr, parent.multi_mesh_manager.map_to_child_handles[2], parent.switch_vertex(edge)));
-
-    CHECK(child0_ptr->switch_edge(edge_child0) == MultiMeshManager::map_tuple_between_meshes(parent, *child0_ptr, parent.multi_mesh_manager.map_to_child_handles[0], parent.switch_edge(edge)));
-    CHECK(child1_ptr->switch_edge(edge_child1) == MultiMeshManager::map_tuple_between_meshes(parent, *child1_ptr, parent.multi_mesh_manager.map_to_child_handles[1], parent.switch_edge(edge)));
-    CHECK(child2_ptr->switch_edge(edge_child2) == MultiMeshManager::map_tuple_between_meshes(parent, *child2_ptr, parent.multi_mesh_manager.map_to_child_handles[2], parent.switch_edge(edge)));
+    for (PrimitiveType pt : {PV, PE}) {
+        CHECK(
+            child0.switch_tuple(edge_child0, pt) ==
+            get_single_child_tuple(child0, parent.switch_tuple(edge, pt)));
+        CHECK(
+            child1.switch_tuple(edge_child1, pt) ==
+            get_single_child_tuple(child1, parent.switch_tuple(edge, pt)));
+        CHECK(
+            child2.switch_tuple(edge_child2, pt) ==
+            get_single_child_tuple(child2, parent.switch_tuple(edge, pt)));
+    }
 }
 
-TEST_CASE("test_split_multi_mesh","[multimesh][2D]")
+/*
+TEST_CASE("test_split_multi_mesh", "[multimesh][2D]")
```

Comment:
Will this test case come back or is it dead code?

Reply:
working on for next pr